packages/xo-server-sdn-controller/src/index.js

```diff
@@ -715,6 +671,70 @@ class SDNController extends EventEmitter {
     }
   }
 
+  async _addHostToPoolNetworks(host) {
+    const xapi = host.$xapi
+
+    const tunnels = filter(xapi.objects.all, { $type: 'tunnel' })
+    for (let i = 0; i < tunnels.length; ++i) {
+      const tunnel = tunnels[i]
+      const accessPIF = await xapi._getOrWaitObject(tunnel.access_PIF)
+      if (accessPIF.host !== host.$ref) {
+        continue
+      }
+
+      const poolNetwork = find(this._poolNetworks, {
+        network: accessPIF.network,
+      })
+      if (poolNetwork == null) {
+        continue
+      }
+
+      if (accessPIF.currently_attached) {
+        continue
+      }
+
+      log.debug(
+        `Pluging PIF: '${accessPIF.device}' for host: '${
+          host.name_label
+        }' on network: '${accessPIF.$network.name_label}'`
+      )
+      try {
+        await xapi.call('PIF.plug', accessPIF.$ref)
+      } catch (error) {
+        log.error(
+          `XAPI error while pluging PIF: '${accessPIF.device}' on host: '${
+            host.name_label
+          }' for network: '${accessPIF.$network.name_label}'`
+        )
+      }
+
+      const starCenter = await xapi._getOrWaitObject(poolNetwork.starCenter)
+      await this._addHostToNetwork(host, accessPIF.$network, starCenter)
+    }
+  }
+
+  async _hostUnreachable(host) {
+    const poolNetworks = filter(this._poolNetworks, { starCenter: host.$ref })
+    for (let i = 0; i < poolNetworks.length; ++i) {
```

Comment:
If the the index of the loop is used only for this purpose const tunnel = tunnels[i] than it's better to use this kind of loop:
```js
for(let tunnel of tunnels) {
...
```
And get rid of const tunnel = tunnels[i]